node/bft/src/sync/mod.rs

```diff
@@ -1252,18 +1253,18 @@ mod tests {
         spawn_blocking!(ledger.advance_to_next_block(&block))?;
 
         // Initialize the syncing ledger.
-        let syncing_ledger = spawn_blocking!(CurrentLedger::load(genesis, StorageMode::new_test(None))).unwrap();
-        let syncing_ledger = Arc::new(CoreLedgerService::new(syncing_ledger, Default::default()));
+        let syncing_ledger = spawn_blocking!(CurrentLedger::load(genesis, StorageMode::Test(None))).unwrap();
+        let syncing_ledger = Arc::new(CoreLedgerService::new(syncing_ledger, SimpleStoppable::new()));
+
         // Initialize the gateway.
-        let storage_mode = StorageMode::new_test(None);
         let gateway = Gateway::new(
             account.clone(),
             storage.clone(),
             syncing_ledger.clone(),
             None,
             &[],
             false,
-            storage_mode,
+            StorageMode::Test(None),
```

Comment:
```suggestion
            StorageMode::new_test(None),
```
this should always be used when a full ledger is involved